src/core/ekat_test_utils.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef EKAT_TEST_UTILS_HPP
+#define EKAT_TEST_UTILS_HPP
+
+#include <map>
+#include <vector>
+#include <string>
+
+namespace ekat {
+
+struct TestSession {
+  static TestSession& get () {
+    static TestSession s;
+    return s;
+  }
+
+  void parse_test_args (const std::vector<char*>& args);
+
+  // These three store test args, but in different ways
+  //  - flags contains args of the form "-s" or "--p" (not followed by another string) as a map string->bool
+  //  - params contains args of the form "-p value"/"--param value"
+  //  - vec_params contains args of the form "-p val1 val2 val3"/"--param val1 val2 val3"
+  // Notice that, as such, params entries will ALWAYS be in vec_params as well (with legnth 1)
+  std::map<std::string,bool>                      flags;
+  std::map<std::string,std::string>               params;
+  std::map<std::string,std::vector<std::string>>  vec_params;
+private:
+  TestSession() = default;
+};
+
+
+template <typename RealType, typename rngAlg, typename PDF>
+void genRandArray(RealType *const x, int length, rngAlg &engine, PDF &&pdf) {
```

Comment:
The overload for kokkos views and packs have been moved in the respective sub-packages